test/LibraryManager.IntegrationTest/LibmanTestsUtility.cs

```diff
@@ -30,6 +29,26 @@ public static void WaitForCompletionEntries(IVisualStudioTextEditorTestExtension
             }
         }
 
+        public static void WaitForCompletionEntry(IVisualStudioTextEditorTestExtension editor, string expectedEntry, bool caseInsensitive, int timeout = 1000)
+        {
+            WaitForCompletionEntries(editor, new[] { expectedEntry }, caseInsensitive, timeout);
+        }
+
+        public static void WaitForCompletionEntryNotPresent(IVisualStudioTextEditorTestExtension editor, string entryText, bool caseInsensitive, int timeout = 1000)
+        {
+            string errorMessage = WaitForCompletionEntriesHelper(editor, new[] { entryText }, caseInsensitive, timeout);
```

Comment:
>string errorMessage = WaitForCompletionEntriesHelper(editor, new[] { entryText }, caseInsensitive, timeout); [](start = 12, length = 108)

So, any call into this is going to have to timeout to succeed? This seems weird.

Reply:
Change the WaitForCompletionEntriesHelper function to not make a waitfor call if the whole completion is already got.

---
In reply to: [193903052](https://github.com/aspnet/LibraryManager/pull/108#discussion_r193903052) [](ancestors = 193903052)